src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,250 @@
+            {
+                header = false;
+                continue;
+            }
+            std::string vehicleType = entry.at(1);
+
+            auto const vehicleId = stoi( entry.at( 0 ) );
+            auto const homeFireStation = findFireStation( stoi( entry.at( 2 ) ) );
+            auto const location = findFireStation(stoi(entry.at(2))).getFireStationLocation();
+            auto const maxNumCrew = stoi(entry.at(3));
+            auto const curNumCrew = stoi(entry.at(4));
+            auto const maxWaterVolumeLitres = stof(entry.at(5));
+            auto const curWaterVolumeLitres = stof(entry.at(6));
+            auto const workFactor = stof(entry.at(7));
+
+            if (vehicleType == "Engine")
+            {
+                districtVehicles.push_back( std::make_unique<FireEngine>(FireEngine(vehicleId, location,
+                                                                                    homeFireStation, maxNumCrew,
+                                                                                    curNumCrew, maxWaterVolumeLitres,
+                                                                                    curWaterVolumeLitres,
+                                                                                    VehicleStatus::Available,
+                                                                                    workFactor)));
+                std::cout << "Adding vehicle #" << vehicleId << " to global list" <<std::endl;
+            }
+            else if (vehicleType == "Ladder")
+            {
+                districtVehicles.push_back( std::make_unique<FireLadder>(FireLadder(vehicleId, location,
+                                                                                    homeFireStation, maxNumCrew,
+                                                                                    curNumCrew, maxWaterVolumeLitres,
+                                                                                    curWaterVolumeLitres,
+                                                                                    VehicleStatus::Available,
+                                                                                    workFactor)));
+                std::cout << "Adding vehicle #" << vehicleId << " to global list" <<std::endl;
+            }
+        }
+    }
+
+    void DistrictResources::processFireStationSampleData(const std::string &fileName)
+    {
+        std::vector<std::vector<std::string>> stations;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            stations.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : stations)
```

Comment:
```suggestion
        for (const std::vector<std::string> & entry : stations)
```
You probably don't want a copy here.